video/out/gl_video.c

```diff
@@ -1693,11 +1702,15 @@ static void pass_delinearize(struct gl_video *p, enum mp_csp_trc trc)
 }
 
 // Takes care of the main scaling and pre/post-conversions
-static void pass_scale_main(struct gl_video *p)
+static void pass_scale_main(struct gl_video *p, struct gl_transform offset)
 {
     // Figure out the main scaler.
     double xy[2];
     get_scale_factors(p, xy);
+    // Adjust the scale factors to account for the pre-scaler. For simplicity
+    // we just disregard shearing (no pre-scaler does shearing anyway)
+    xy[0] /= offset.m[0][0];
+    xy[1] /= offset.m[1][1];
```

Comment:
I would like to refactor the way scaling is done in mpv in general. Most of the confusing logic comes from the fact that we have src_rect, dst_rect and weird stuff like that.

Instead of using those, I would immediately transform them into our own struct gl_transform because those are self-contained and easy to compose, apply, etc.

Most of the logic inside pass_scale_main, blend_subs etc. would get simpler and more resistant to bugs like these.